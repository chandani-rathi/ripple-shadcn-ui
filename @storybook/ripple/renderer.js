// renderer.js
export async function renderToCanvas({ storyFn, canvasElement }) {
  console.log("renderer")
  const output = storyFn();

  // Example: simple HTML render
  if (typeof output === 'string') {
    canvasElement.innerHTML = output;
  } else if (output instanceof HTMLElement) {
    canvasElement.innerHTML = '';
    canvasElement.appendChild(output);
  } else {
    canvasElement.innerHTML = `<pre>Unsupported output: ${JSON.stringify(output)}</pre>`;
  }
}
