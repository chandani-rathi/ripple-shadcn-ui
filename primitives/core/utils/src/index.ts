export * from './utils.ripple';
export * from "./normalize-css-property-value"