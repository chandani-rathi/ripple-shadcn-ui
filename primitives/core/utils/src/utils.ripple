
const NAME = 'Utils';

export component Utils(props){
  <props.@children />
}


const Root = Utils

export {
  Root
}