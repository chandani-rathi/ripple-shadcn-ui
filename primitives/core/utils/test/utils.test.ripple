import { axe } from 'vitest-axe';
import userEvent from '@testing-library/user-event';
import { Utils }  from "../src/utils.ripple";

const TEXT_CHILD = "TEST"

describe("utils", () => {
    beforeEach(() => {
		component Test(){
			<Utils>{TEXT_CHILD}</Utils>
		}
		render(Test);
	});

    it('should have no accessibility violations', async () => {
		expect(await axe(container)).toHaveNoViolations();
	});
    
    it("should work", () => {
         expect(container).toMatchSnapshot()
    })
})