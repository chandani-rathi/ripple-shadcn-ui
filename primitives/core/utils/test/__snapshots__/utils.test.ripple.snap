// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`utils > should work 1`] = `
<div>
  <!---->
  <!---->
  TEST
  <!---->
  <!---->
  
</div>
`;
